### ISSUE TITLE
runtime-rs: network cleanup can leave Tokio workers in the pod namespace

### ISSUE THREAD

**Issue by realsdx** [kata-containers/kata-containers#13885]
## Description of problem

`ResourceManagerInner::cleanup()` directly awaits `NetworkWithNetns::remove()`,
which holds `NetnsGuard` across `endpoint.detach().await`. `setns()` changes an
OS thread's namespace, but async tasks can yield or move between threads.
Cleanup can therefore restore the namespace on the wrong worker, or allow
unrelated tasks to use the pod network while detach is suspended.

Use a disposable node with working containerd/CNI, a runtime-rs handler,
`crictl`, `strace`, and root/ptrace access. The steps below reproduced
wrong-thread restoration on Azure Linux with runtime-rs 3.32.0, reported commit
`93b0c69435add9b52e3d272b6cab2f60f6eaac7d`.

1. Create a sandbox using the installed runtime-rs handler:

   ```bash
   runtime_handler=kata-clh-runtime-rs # replace with the configured handler
   cat >/tmp/kata-netns-check.json <<'EOF'
   {
     "metadata": {"name": "netns-check", "namespace": "default", "uid": "netns-check", "attempt": 0},
     "log_directory": "/tmp",
     "linux": {}
   }
   EOF
   sandbox_id=$(sudo crictl runp --runtime "$runtime_handler" /tmp/kata-netns-check.json)
   sudo pgrep -af "containerd-shim.*${sandbox_id}"
   ```

2. Identify that sandbox's shim PID from the output. In another terminal,
   attach before stopping the sandbox:

   ```bash
   shim_pid=12345 # replace with this sandbox's actual shim PID
   sudo strace -f -tt -yy -e trace=setns,socket,connect \
     -p "$shim_pid" -o /tmp/kata-netns-teardown.strace
   ```

3. In the first terminal, trigger normal teardown:

   ```bash
   sudo crictl stopp "$sandbox_id"
   sudo crictl rmp "$sandbox_id"
   ```

   The tracer normally exits with the shim; otherwise detach with Ctrl-C.
   Retain the trace, shim logs, and source revision for each build.

## Expected result

Namespace entry, detach, and restoration stay on one isolated thread without
affecting unrelated Tokio tasks. Normal sandbox teardown succeeds.

## Actual result

Sandbox creation, stop, and removal succeeded, but the trace showed:

```text
25569 12:49:55.085490 setns(16</run/netns/cni-1e95b3c2-0529-2b3f-9025-56ddae0ed07d>, CLONE_NEWNET) = 0
25568 12:49:55.087144 setns(13<net:[4026531840]>, CLONE_NEWNET) = 0
```

TID **25569** entered the pod namespace; **25568** restored the original
namespace. No restoration on 25569 appeared before it exited.

This confirms wrong-thread restoration on the installed build, not a captured
connection failure or a before/after test of the fix. Reproduction depends on
scheduling; a clean run does not rule it out.

### How we found this

Missing traces during OTLP development led us to teardown. After correcting
tracer shutdown ordering, seven of 16 exports still failed to reach the
host-local collector. Isolating network cleanup restored delivery to 16/16
with that ordering unchanged. This development-branch result prompted the
namespace investigation.
